### ISSUE TITLE
Use shared libraries for linking the main executable

### ISSUE THREAD

**Issue by moubctez** [google/brotli#751]
Currenly, the main executable is quite large, because it is being linked against static libraries. Please, consider linking it against the shared libraries. The following patch does the shared linking:
```
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -183,7 +183,7 @@ endif()
 
 # Build the brotli executable
 add_executable(brotli ${BROTLI_CLI_C})
-target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+target_link_libraries(brotli ${BROTLI_LIBRARIES})
 
 # Installation
 if(NOT BROTLI_BUNDLED_MODE)
```

**Comment by eustas** [comment]
That is done intentionally, so that brotli executable compiled in one container would be executed in other container without any deps.

Do you have some specific scenario, where shared-linked brotli CLI would be useful?
In that case it would be better to add a configuration option...

**Comment by lazka** [comment]
Looks like this was fixed in 641bec0e30bea648b3, the main binary now links against the shared lib by default with v1.1.0
